modules/library/main/src/main/java/org/geotools/filter/function/JsonFunctionUtils.java

```diff
@@ -0,0 +1,95 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2022, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.filter.function;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import com.fasterxml.jackson.core.JsonTokenId;
+import java.io.IOException;
+
+/** Utility methods for serialize JSON array and object as String */
+interface JsonFunctionUtils {
```

Comment:
Making these utils methods on an interface makes the default methods public.  Looks like the methods are internal tooling, maybe could be better to use an abstract class with protected methods or a package protected class and call it as dependency?

Reply:
Well, this is an interesting one :scream::
- As someone who did Scala, I understand the functional approach, 
- as a Java developer, indeed the abstract class or Utils class with static methods is the way to go.

I think Java wins here.